### PR TITLE
Sscsci 1123 international bulk print

### PR DESCRIPTION
**Pull request by Tim97eng**
### Jira link

See [SSCSCI-1123](https://tools.hmcts.net/jira/browse/SSCSCI-1123)

### Change description

- Introducing the ability to add isInternational flag for bulk print
- Introduced new complexType additions for Address and Identity

Re-work from initial PR that will now:

- hide IBA based fields for dwpUploadResponse for UC cases and addHearing event.
- Remove the label of MRN/IRN that was on manage case UI until PO approval and confirmation

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by nilay913 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java [hunk 1]
```diff
@@ -161,6 +164,11 @@ private static Map<String, Object> getAdditionalData(final SscsCaseData sscsCase
         additionalData.put(CASE_IDENTIFIER, sscsCaseData.getCcdCaseId());
         additionalData.put(APPELLANT_NAME, sscsCaseData.getAppeal().getAppellant().getName().getFullNameNoTitle());
         additionalData.put(RECIPIENTS, getRecipients(recipient));
+
+        YesNo isInUk = sscsCaseData.getAppeal().getAppellant().getAddress().getInMainlandUk();
+        if (isInUk != null && !isYes(isInUk)) {
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java:169]: I don't think you need the null check here

**Review by benouaer — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java (hunk 1 above)
Comment (Tim97eng) [reply src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java:169]: The null check was added for the cases that aren't IBA or dont set this field

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java:169]: null check is not required as the two methods are fine if you pass null values:
`    public static boolean isYes(YesNo yesNo) {
        return YES.equals(yesNo);
    }

    public static boolean isYes(String yesNo) {
        return nonNull(yesNo) && YES.getValue().equalsIgnoreCase(yesNo);
    }
`

**Review by benouaer — APPROVED** [review]

File: definitions/benefit/sheets/CaseEventToComplexTypes/CaseEventToComplexTypes.json [hunk 2]
```diff
@@ -1492,6 +1503,159 @@
...
+    "DisplayContext": "OPTIONAL",
+    "FieldShowCondition": "appeal.benefitType.code=\"infectedBloodAppeal\"",
+    "LiveFrom": "01/01/2018",
+    "FieldDisplayOrder": 11,
+    "PageID": "1.0"
+  },
+  {
+    "CaseEventID": "addHearing",
+    "CaseFieldID": "hearings",
+    "ListElementCode": "venue.address.portOfEntry",
```
Comment (tomxelliott) [inline definitions/benefit/sheets/CaseEventToComplexTypes/CaseEventToComplexTypes.json:1640]: this one shouldn't be here as we populate it in the backend. please remove this field from the event

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java (hunk 1 above)
Comment (Tim97eng) [reply src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/BulkPrintService.java]: Changed the code to be clearer with what I intended it to do when handling non-iba cases

**Review by tomxelliott — APPROVED** [review]

**Review by benouaer — APPROVED** [review]

**Review by tomxelliott — APPROVED** [review]

**Review by benouaer — APPROVED** [review]

**Pull request merged**